File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        //store the count
        unordered_map<int,int> count;
        for(int i = 0;i<nums.size();i++){
            count[nums[i]] += 1;
        }
        //create a list of the pair in count
        vector<pair<int,int>> arr;
        for(const auto& p : count){
            arr.push_back({p.second, p.first});
        }
        //sort the pair in descending order
        sort(arr.rbegin(), arr.rend());

        //store top 2 number in ans list
        vector<int> ans;
        for(int i = 0; i< k; i++){
            ans.push_back(arr[i].second);
        }
        return ans;
    }
};

File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-1.java
class Solution {
    public int removeDuplicates(int[] nums) {
        if(nums == null || nums.length == 0) {
            return 0;
        }
        if(nums.length ==1) {
            return 1;
        }
        int ans = 1;
        int last = 1;
        for (int i=1;i<nums.length;i++) {
            if(nums[i] != nums[i-1]) {
                nums[last] = nums[i];
                last = last + 1;
                ans = last;
            }
        }
        return ans;
    }
}